src/test/streams_tests.cpp

```diff
@@ -202,6 +202,16 @@ BOOST_AUTO_TEST_CASE(streams_serializedata_xor)
             std::string(ds.begin(), ds.end()));
 }
 
+#define BOOST_CHECK_THROW_WHAT(statement, exception, str) \
+    do { \
+        try { \
+            statement; \
+            BOOST_CHECK(false); \
+        } catch (const exception& e) { \
+            BOOST_CHECK(strstr(e.what(), str) != nullptr); \
+        } \
+    } while (false)
+
```

Comment:
No functional change here, just refactoring a check that's repeated 4 times. (It's a macro rather than a function so that the line numbers report correctly.) This macro is misnamed (since it's not part of Boost), but it's so close to `BOOST_CHECK_THROW()` that this is probably the least confusing name. (But I can change it.)

Reply:
Update: I removed this new test macro, `BOOST_CHECK_THROW_STR()`, because it's out of scope for this PR.